src/realm/util/logger.cpp

```diff
@@ -112,6 +112,8 @@ const std::string_view Logger::level_to_string(Level level) noexcept
 
 void StderrLogger::do_log(Level level, const std::string& message)
 {
+    static Mutex mutex;
+    LockGuard l(mutex);
```

Comment:
StderrLogger should be wrapped with ThreadSafeLogger when this behavior is desired rather than adding it to each logger.

Reply:
Sure, i'll try to revert this back on next pr in order to not let this one to stale for longer. It's just that wrapping StderrLogger is not the only thing. The very same instance should be then shared everywhere on every config. It's not the case now. Also, the problem is with a few places where logger is initialized from realm default, which doesn't know anything about other parts. So, it's easy to misuse as it turned out during these fixes here.